### PR TITLE
Structured interfaces

### PR DESCRIPTION
**Pull request by dannystey**
- added the defines.ts for interfaces of props / state and methods
- added a method object to capsulate methods which could be used in template
- concat props, state and methods to data and deliver them to the template

### CODE REVIEW

**Comment by jurekbarth** [comment]
That adds a lot of boilerplate if i create a new component. If we could add an option for that, i would be happier :)

**Comment by jurekbarth** [comment]
@dannystey @SheepFromHeaven how do we proceed here?

**Comment by SheepFromHeaven** [comment]
@dannystey I would like to remove the example code, then we can merge

**Comment by timomayer** [comment]
@dannystey?

File: _templates/component/with-prompt/prompt.js [hunk 1]
```diff
@@ -19,6 +19,13 @@ const questions = [
     },
     message: 'Component name 🙏'
   },
+  {
+    type: 'checkbox',
+    name: 'generateParts',
+    message: 'Which features your component will use?',
```
Comment (SheepFromHeaven) [inline _templates/component/with-prompt/prompt.js]: This not is english!

**Pull request merged**